src/core/ServiceRegistry.ts

```
@@ -1,74 +1,112 @@
+export class ServiceRegistry {
+  private static logger = createRichLogger('ServiceRegistry');
+  private static services = new Map<string, ServiceDefinition>();
+  private static isInitialized = false;
 
-  register<T>(definition: ServiceDefinition<T>): void {
-    if (this.services.has(definition.name)) {
-      throw new Error(`Service ${definition.name} is already registered`);
-    }
-    this.services.set(definition.name, definition);
+  static initialize(): void {
+    if (this.isInitialized) return;
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)